lib/task_listener.js

```diff
@@ -164,7 +168,8 @@ export default class TaskListener extends EventEmitter {
     if (exceedsThreshold) return;
 
     let claims = await this.taskQueue.claimWork(availableCapacity);
-    claims.forEach(this.runTask.bind(this));
+    let tasksets = await Promise.all(claims.map(this.coalesceClaim.bind(this)));
+    await Promise.all(tasksets.map(this.runTaskset.bind(this)));
```

Comment:
In the old code, I think most things in runTask was wrapped in a try/catch and would result in a "task error" event if something was caught.  That was how it was handled before unless there is a particular case where you see an error that wasn't being handled there.

We've done elsewhere somewhere like attaching .catch() on each promise returned and appending the error to a list for it to be logged rather than it throwing an uncaught error or calling reject() on the promise.

Something like this:
https://github.com/taskcluster/docker-worker/blob/49056e24fab16c63979bbad546ccf63fd809ffa1/lib/features/docker_save.js#L92